marquez_airflow/dag.py

```diff
@@ -117,74 +117,87 @@ def create_dagrun(self, *args, **kwargs):
             dagrun = args[0]
             self.log.debug(f"handle_callback() dagrun : {dagrun}")
             task_instances = dagrun.get_task_instances()
-            self.log.info(f"{task_instances}")
+
             for ti in task_instances:
-                task = dagrun.get_dag().get_task(ti.task_id)
-                self.log.info(f"ti: {ti} of task: {task}")
+                ti.task = self.get_task(ti.task_id)
+                task = ti.task
 
                 extractor = self._get_extractor(task)
 
-                ti_location = self._get_location(task)
+                marquez_job_run_ids = self._get_run_ids(ti,
+                                                        dagrun.run_id,
+                                                        kwargs.get('session'))
 
                 if extractor:
-
                     steps_meta = add_airflow_info_to(
                         task,
                         extractor(task).extract_on_complete(ti))
-
                     for step in steps_meta:
-                        self.log.info(f'step: {step}')
-
-                        marquez_run_id = self._get_marquez_run_id(
-                            ti, dagrun, kwargs)
-                        self.log.info(f'marquez_run_id: {marquez_run_id}')
-
-                        self.register_datasets(step.inputs)
-                        self.register_datasets(step.outputs, marquez_run_id)
-                        inputs = self._to_dataset_ids(step.inputs)
-                        outputs = self._to_dataset_ids(step.outputs)
-                        self.log.info(
-                            f'inputs: {inputs} '
-                            f'outputs: {outputs} '
-                        )
-                        self.get_or_create_marquez_client().create_job(
-                            namespace_name=self.marquez_namespace,
-                            job_name=step.name,
-                            job_type=JobType.BATCH,
-                            location=step.location or ti_location,
-                            input_dataset=inputs,
-                            output_dataset=outputs,
-                            description=self.description,
-                            context=step.context,
-                            run_id=marquez_run_id)
-
-                        self.log.info(f"client.create_job(run_id="
-                                      f"{marquez_run_id}) successful.")
-
-                try:
-                    self.report_jobrun_change(
-                        ti, dagrun.run_id, **kwargs)
-                except Exception as e:
-                    self.log.error(
-                        f'Failed to record task run state change: {e} '
-                        f'dag_id={self.dag_id}',
-                        exc_info=True)
 
+                        # No run associated, create backdated run & job record
+                        if not marquez_job_run_ids:
+                            self._alternate_create_flow(step, ti)
```

Comment:
instead of `_alternate_create_flow`, should we call it something more explicit like `_create_flow_on_completion`?